### ISSUE TITLE
_.assign is not a function

### ISSUE THREAD

**Issue by omouse** [samselikoff/ember-cli-mirage#413]
There's an issue with lodash not being loaded up or included, either in the bower_components or node_modules, I'm not sure which.

We're using version 0.11.1 of ember-cli-mirage.

```
TypeError: _.assign is not a function
    at Function.exports.default [as extend] (http://localhost:4201/assets/vendor.js:159193:7)
    at http://localhost:4201/assets/vendor.js:158829:46
    at mod.state (http://localhost:4201/assets/vendor.js:150:29)
    at tryFinally (http://localhost:4201/assets/vendor.js:30:14)
    at requireModule (http://localhost:4201/assets/vendor.js:148:5)
    at requireFrom (http://localhost:4201/assets/vendor.js:121:12)
    at reify (http://localhost:4201/assets/vendor.js:106:22)
    at mod.state (http://localhost:4201/assets/vendor.js:149:17)
    at tryFinally (http://localhost:4201/assets/vendor.js:30:14)
    at requireModule (http://localhost:4201/assets/vendor.js:148:5)
```

Here's how our dependencies in package.json looks:

```
"devDependencies": {
    "body-parser": "^1.2.0",
    "bower": "^1.6.3",
    "broccoli-asset-rev": "^2.0.2",
    "broccoli-concat": "0.0.13",
    "broccoli-merge-trees": "^0.2.1",
    "broccoli-replace": "^0.1.7",
    "broccoli-static-compiler": "0.2.1",
    "broccoli-svgstore": "^0.1.0",
    "chalk": "0.4.0",
    "ember-cli": "0.2.5",
    "ember-cli-app-version": "0.3.3",
    "ember-cli-async-button": "^0.3.1",
    "ember-cli-autoprefixer": "^0.3.0",
    "ember-cli-babel": "^5.0.0",
    "ember-cli-chartist": "^0.2.9",
    "ember-cli-coffeescript": "~0.0.7",
    "ember-cli-dependency-checker": "^1.0.0",
    "ember-cli-htmlbars": "0.7.6",
    "ember-cli-ic-ajax": "0.1.1",
    "ember-cli-inject-live-reload": "^1.3.0",
    "ember-cli-mirage": "0.1.11",
    "ember-cli-pretender": "0.0.5",
    "ember-cli-qunit": "0.3.13",
    "ember-cli-sass": "^4.0.0",
    "ember-cli-showdown": "2.1.4",
    "ember-cli-simple-auth": "0.6.7",
    "ember-cli-simple-auth-oauth2": "0.6.7",
    "ember-cli-uglify": "^1.2.0",
    "ember-cli-zero-clipboard": "^1.0.2",
    "ember-data": "1.0.0-beta.17",
    "ember-disable-proxy-controllers": "^0.7.0",
    "ember-export-application-global": "^1.0.2",
    "ember-list-view": "0.0.6",
    "ember-multiselect-checkboxes": "^0.3.1",
    "ember-select-2": "^1.2.0",
    "ember-sortable": "1.6.2",
    "express": "^4.8.5",
    "glob": "^4.4.0",
    "grunt": "^0.4.5",
    "grunt-aws-s3": "^0.9.4",
    "http-proxy": "^1.1.6",
    "mirage": "0.0.5",
    "node-sass": "^3.3.2",
    "originate": "0.1.5",
    "sync-exec": "^0.5.0"
  }
```

Here's how our dependencies in bower.json looks:

```
"dependencies": {
    "accounting": "~0.3.2",
    "animate.css": "~3.2.0",
    "autofill-event": "~1.0.0",
    "bootstrap": "~3.2.0",
    "chartkick": "~1.3.0",
    "dropzone": "uniiverse/dropzone#3.10.3",
    "ember-simple-auth": "0.6.7",
    "ember-validations-shiv": "uniiverse/ember-validations-shiv",
    "highcharts": "~4.1.4",
    "intercom": "edude03/intercom.js#fix-intercom-with-bower",
    "jquery-color": "~2.1.2",
    "jquery-form": "~3.46.0",
    "jquery.payment": "~1.2.2",
    "moment": "~2.7.0",
    "papaparse": "~4.1.0",
    "select2": "3.5.2",
    "select2-bootstrap-css": "~1.4.6",
    "sockjs": "~0.3.4",
    "svg4everybody": "~1.0.0",
    "thousandwords": "0.1.9",
    "underscore": "~1.7.0",
    "underscore.inflection": "~1.1.1",
    "underscore.string": "~2.3.3",
    "ember": "1.12.0",
    "ember-cli-shims": "ember-cli/ember-cli-shims#0.0.3",
    "ember-cli-test-loader": "ember-cli-test-loader#0.1.3",
    "ember-data": "1.0.0-beta.18",
    "ember-load-initializers": "ember-cli/ember-load-initializers#0.1.4",
    "ember-qunit": "0.3.3",
    "ember-qunit-notifications": "0.0.7",
    "ember-resolver": "~0.1.15",
    "jquery": "^1.11.1",
    "loader.js": "ember-cli/loader.js#3.2.0",
    "qunit": "~1.17.1",
    "jquery.easing": "~1.3.1",
    "chartist": "~0.9.4",
    "jquery.kinetic": "~2.1.0",
    "keen-js": "3.2.7",
    "jquery-ui": "~1.11.4",
    "pretender": "~0.10.1",
    "lodash": "~3.7.0",
    "Faker": "~3.0.0"
  }
```

In a separate project using v0.11.1 I was able to get the tests running without this error message. It's only in my particular project where I get this "_.assign is not a function" error message. I'm pretty sure it's a dependency or build issue.

**Comment by davidpatters0n** [comment]
+1

**Comment by omouse** [comment]
might be fixed by #322 
suggest that the next version of ember-cli-mirage be pushed to npm :cry:

**Comment by nolaneo** [comment]
:+1: Also seeing this

**Comment by samselikoff** [comment]
I'll look at this tonight

**Comment by nolaneo** [comment]
This does appear to be fixed by switching to the beta version, so I echo @omouse that getting a new version onto NPM might be nice :grin:

**Comment by samselikoff** [comment]
https://github.com/samselikoff/ember-cli-mirage/commit/678c2b8285211d19efae259af205203c66d73094 is the fix. I'm trying to figure out the best course of action. It's clear to me I should have been doing 0.2.x development in a branch - this seems obvious, but when I started everything in 0.2.x was backwards compatible. At some point, I decided to make some breaking changes, and then I should have started working on a separate branch.

Now this commit is in master but unavailable to 0.1.x series. I could make an `0-1-stable` branch of development (like AMS does), and have people install from that. But then will I have to keep that branch around forever (since those commits won't be in master...)? That seems like a bad option.

The other alternative is, I could publish 0.1.12 from the above commit. There is one "breaking change", moving `/app/mirage` to `/mirage`, but even that is configurable if you wanted to keep it in `/app/mirage`. Then, we wouldn't need another branch, and all tags would stay in master (where they belong).

What do you all think? (And, lesson learned - always do new unfinished development on a branch!)

**Comment by samselikoff** [comment]
Can I see your app's Mirage code that's triggering this bug?

**Comment by bekzod** [comment]
sorry I might be out of context but why not to use `Ember.assign`?

**Comment by samselikoff** [comment]
Mirage doesn't actually use any functionality from Ember, eventually the goal is to extract it into a bower package. It doesn't know the app making the requests is an Ember app.

**Comment by kellyselden** [comment]
Tried to add mirage to a project today and got this error.

**Comment by samselikoff** [comment]
@kellyselden what version of Mirage?

**Comment by kellyselden** [comment]
@samselikoff The latest on 0.1.x. I'm mobile right now but pretty sure it was 0.1.11.

**Comment by kellyselden** [comment]
@samselikoff I think you can branch off 0.1.x, tag it, then delete the branch and have the tag remain. I've done that on a project before for a hotfix.

**Comment by samselikoff** [comment]
0.12 released to address this: https://github.com/samselikoff/ember-cli-mirage/releases/tag/v0.1.12

**Comment by jrjohnson** [comment]
I'm not longer able to build my app after this push (0.2.12).  Getting 'Cannot read property 'path' of undefined'  https://travis-ci.org/#L1902 
Maybe something about using NPM v3?

**Comment by samselikoff** [comment]
Looking into it

**Comment by samselikoff** [comment]
Should be fixed with [0.1.13](https://github.com/samselikoff/ember-cli-mirage/releases/tag/v0.1.13)
